applications/ChimeraApplication/custom_processes/apply_chimera_process_fractional_step.h

```diff
@@ -0,0 +1,371 @@
+     * @param rBoundaryModelPart The boundary modelpart for which the continuity is to be enforced.
+     * @param pBinLocator The bin based locator formulated on the background. This is used to locate nodes on rBoundaryModelPart.
+     */
+    void ApplyContinuityWithMpcs(ModelPart &rBoundaryModelPart, PointLocatorPointerType &pBinLocator) override
+    {
+        //loop over nodes and find the triangle in which it falls, then do interpolation
+        MasterSlaveContainerVectorType velocity_ms_container_vector;
+        MasterSlaveContainerVectorType pressure_ms_container_vector;
+#pragma omp parallel
+        {
+            const IndexType num_constraints_per_thread = (rBoundaryModelPart.NumberOfNodes() * 4)/omp_get_num_threads();
+#pragma omp single
+            {
+                velocity_ms_container_vector.resize(omp_get_num_threads());
+                for (auto &container : velocity_ms_container_vector)
+                    container.reserve(num_constraints_per_thread);
+
+                pressure_ms_container_vector.resize(omp_get_num_threads());
+                for (auto &container : pressure_ms_container_vector)
+                    container.reserve(num_constraints_per_thread);
+            }
+        }
+        std::vector<int> constraints_id_vector;
+
+        int num_constraints_required = (TDim + 1) * (rBoundaryModelPart.Nodes().size());
+        BaseType::CreateConstraintIds(constraints_id_vector, num_constraints_required);
+
+        const int max_results = 10000;
+        const IndexType n_boundary_nodes = rBoundaryModelPart.Nodes().size();
+        IndexType counter = 0;
+        IndexType removed_counter = 0;
+        IndexType not_found_counter = 0;
+
+        for (IndexType i_bn = 0; i_bn < n_boundary_nodes; ++i_bn)
+        {
+            ModelPart::NodesContainerType::iterator i_boundary_node = rBoundaryModelPart.NodesBegin() + i_bn;
+            Node<3>::Pointer p_boundary_node = *(i_boundary_node.base());
+
+            BaseType::mNodeIdToConstraintIdsMap[p_boundary_node->Id()].reserve(150);
+        }
+
+        BuiltinTimer loop_over_b_nodes;
+
+#pragma omp parallel for shared(constraints_id_vector, velocity_ms_container_vector, pressure_ms_container_vector, pBinLocator) reduction(+                                                             \
+                                                                                                                                          : not_found_counter) reduction(+                              \
+                                                                                                                                                                         : removed_counter) reduction(+ \
+                                                                                                                                                                                                      : counter)
+        for (IndexType i_bn = 0; i_bn < n_boundary_nodes; ++i_bn)
```

Comment:
IndexType is a typedef for std::size_t
https://github.com/KratosMultiphysics/Kratos/blob/6bc6a85fc07acbd745a0ff721fe18ee7299e008f/kratos/includes/model_part.h#L103

For windows compilation, an **signed int** is required for omp parallel for

This is found several times in this PR

Reply:
Thank you for pointing out. I will change them.